### ISSUE TITLE
can we add no unused inline style?

### ISSUE THREAD

**Issue by craigcosmo** [yannickcr/eslint-plugin-react#2310]
```js
const style = {
	header:{
		margin:10
	},
    text:{
    	color: '#000'
    }
}
function App() {
	return (
		<div style={style.header}>one</div>
	);
}
// this will trigger error because text is not used
```

```js
const style = {
	header:{
		margin:10
	},
    text:{
    	color: '#000'
    }
}
function App() {
	return (
		<div style={style.header}>
			<div style={style.text}>one</div>
		</div>
	);
}
// this will not trigger error
```

**Comment by ljharb** [comment]
eslint already reports unused variables; i'm not sure what we'd report for.

**Comment by craigcosmo** [comment]
not unused style, it aint

**Comment by ljharb** [comment]
Right but the error here isn’t unused style, it’s an unused variable. There’s nothing about creating an object named “style” that would necessitate it being accessed in a style prop.

**Comment by craigcosmo** [comment]
eslint react native got a rule like I mention

https://github.com/Intellicode/eslint-plugin-react-native/blob/master/docs/rules/no-unused-styles.md

**Comment by golopot** [comment]
This use case appears to be great fit for [unicorn/no-unused-properties](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/docs/rules/no-unused-properties.md).

**Comment by ljharb** [comment]
React Native is very different than React in this regard, but either way, that shouldn't be its own rule - `no-unused-vars` already covers this.
